extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergManualRefreshTableLocationProvider.java

```diff
@@ -0,0 +1,124 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.layout;
+
+import io.deephaven.engine.table.impl.locations.*;
+import io.deephaven.engine.table.impl.locations.impl.TableLocationFactory;
+import io.deephaven.engine.table.impl.locations.impl.TableLocationKeyFinder;
+import io.deephaven.iceberg.util.IcebergCatalogAdapter;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * <p>
+ * Manually refreshing {@link TableLocationProvider} implementation that delegates {@link TableLocationKey location key}
+ * discovery to a {@link TableLocationKeyFinder} and {@link TableLocation location} creation to a
+ * {@link TableLocationFactory}.
+ * </p>
+ */
+public class IcebergManualRefreshTableLocationProvider<TK extends TableKey, TLK extends TableLocationKey>
+        extends IcebergTableLocationProviderBase<TK, TLK> {
+
+    private static final String IMPLEMENTATION_NAME = IcebergManualRefreshTableLocationProvider.class.getSimpleName();
+
+    public IcebergManualRefreshTableLocationProvider(
+            @NotNull final TK tableKey,
+            @NotNull final IcebergBaseLayout locationKeyFinder,
+            @NotNull final TableLocationFactory<TK, TLK> locationFactory,
+            @NotNull final IcebergCatalogAdapter adapter,
+            @NotNull final TableIdentifier tableIdentifier) {
+        super(tableKey, locationKeyFinder, locationFactory, null, true, adapter, tableIdentifier);
+    }
+
+    // ------------------------------------------------------------------------------------------------------------------
+    // AbstractTableLocationProvider implementation
+    // ------------------------------------------------------------------------------------------------------------------
+
+    @Override
+    public String getImplementationName() {
+        return IMPLEMENTATION_NAME;
+    }
+
+    @Override
+    public synchronized void refresh() {
```

Comment:
This should not be a UOE. If a user creates a static table using this TLP implementation, that should work.

I would actually expect this method to be where we load partitions from the initial snapshot. If that's not so, it should just be a no-op.

Reply:
How would the user create a static table using this? We would use `IcebergStaticTableLocationProvider`

Reply:
That goes back to "if we make a TableDataService around this" we have to support static tables using ticking TLPs. Same reason we need to move the snapshot atomicity up to ATLP, rather than down in the subscription buffer, although we could argue we don't need it in Iceberg at this point.

Reply:
Maybe this is too forward-looking. I guess we're unlikely to use a manual update strategy if we were ever to put these in a TLS anyway. That said, I think having this method call `refreshSnaphsot()`  would be a harmless way to future-proof it. See other comment regarding making sure `refreshSnapshot()` avoids redundant work, since you call it from `activateUnderlyingDataSource()`.